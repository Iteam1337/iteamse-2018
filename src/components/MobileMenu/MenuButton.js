// @flow

import React from 'react'
import styled, { css } from 'styled-components'

type Props = {
  open: boolean,
  toggle: () => void,
}

const Button = styled.button`
  -moz-appearance: none;
  -webkit-appearance: none;
  background-color: #fff;
  border: 0;
  bottom: 20px;
  border-radius: 100%;
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);
  height: 60px;
  position: fixed;
  right: 20px;
  z-index: 2;
  width: 60px;

  &:focus {
    outline: none;
  }
`

const Bar = styled.div`
  height: 2px;
  width: 25px;
  background-color: #000;
  margin: 5px auto;
  transition: all 0.3s ease-in-out;
`

const Hamburger = styled.div`
  ${({ open }) =>
    open &&
    css`
      ${Bar}:first-child {
        transform: translateY(7px) rotate(45deg);
      }

      ${Bar}:nth-child(2) {
        opacity: 0;
      }

      ${Bar}:last-child {
        transform: translateY(-7px) rotate(-45deg);
      }
    `};
`

const MenuButton = ({ open, toggle }: Props) => {
  return (
    <Button data-test="btn-mobile-menu" onClick={toggle}>
      <Hamburger open={open}>
        <Bar />
        <Bar />
        <Bar />
      </Hamburger>
    </Button>
  )
}

export default MenuButton
