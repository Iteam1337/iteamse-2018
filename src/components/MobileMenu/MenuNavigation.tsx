import React from 'react'
import { NavLink } from 'react-router-dom'
import styled, { keyframes } from 'styled-components'

interface MenuNavigationProps {
  open: boolean
}

const StyledLink = styled(NavLink)`
  color: ${({ theme }) => theme.colors.cornflowerBlue};
  font-size: 24px;
  font-weight: 300;
  text-decoration: none;

  &:not(:last-child) {
    margin-bottom: 30px;
  }
`

const fadeIn = keyframes`
  from {
  opacity: 0;
  }

  to {
  opacity: 1;
  }
`

const Menu = styled.div`
  align-items: flex-end;
  animation: ${fadeIn} 150ms ease-in-out 1;
  background-color: rgba(255, 255, 255, 0.93);
  bottom: 0;
  display: grid;
  left: 0;
  padding: 20px 20px 100px;
  position: fixed;
  right: 0;
  text-align: right;
  top: 0;
  z-index: 1;
`

const MenuInner = styled.div`
  display: flex;
  flex-direction: column;

  @media only screen and (device-width: 375px) and (device-height: 812px) and (-webkit-device-pixel-ratio: 3) {
    padding-top: 80px;
  }
`

const MenuNavigation: React.SFC<MenuNavigationProps> = ({ open }) => {
  if (!open) {
    return null
  }

  return (
    <Menu data-test="mobile-menu">
      <MenuInner>
        <StyledLink activeClassName="active-nav" to="/case">
          Case
        </StyledLink>
        <StyledLink activeClassName="active-nav" to="/erbjudanden/ai">
          AI
        </StyledLink>
        <StyledLink activeClassName="active-nav" to="/hur-vi-jobbar">
          Metod
        </StyledLink>
        <StyledLink activeClassName="active-nav" to="/teamet">
          Medarbetare
        </StyledLink>
        <StyledLink activeClassName="active-nav" to="/jobba-hos-oss">
          Karri√§r
        </StyledLink>
        <StyledLink activeClassName="active-nav" to="/om-oss">
          Om
        </StyledLink>
      </MenuInner>
    </Menu>
  )
}

export default MenuNavigation
