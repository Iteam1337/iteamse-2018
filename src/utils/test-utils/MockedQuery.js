import * as React from 'react'
import { ApolloProvider } from 'react-apollo'
import ApolloClient from 'apollo-client'
import { createHttpLink } from 'apollo-link-http'
import createMockedNetworkFetch from 'apollo-mocknetworkinterface'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { BrowserRouter } from 'react-router-dom'
import { ThemeProvider } from 'styled-components'
import { theme } from '../../theme'

type P = {
  children: React.Node,
  loading?: boolean,
  response: Object,
}

const MockedQuery = ({ children, loading, response = {} }: P) => {
  const createResponse = () => {
    return {
      loading: !!loading,
      data: {
        ...response,
      },
    }
  }

  const mockedNetworkFetch = createMockedNetworkFetch(createResponse, {
    latency: 0,
  })

  const client = new ApolloClient({
    link: createHttpLink({
      uri: 'http://localhost:3000',
      fetch: mockedNetworkFetch,
    }),
    cache: new InMemoryCache(),
  })

  return (
    <ApolloProvider client={client}>
      <ThemeProvider theme={theme}>
        <BrowserRouter>{children}</BrowserRouter>
      </ThemeProvider>
    </ApolloProvider>
  )
}

export default MockedQuery
