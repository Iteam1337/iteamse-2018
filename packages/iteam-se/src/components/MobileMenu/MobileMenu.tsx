import * as React from 'react'
import { RouteComponentProps, withRouter } from 'react-router-dom'
import styled from '../../theme'
import MenuButton from './MenuButton'
import MenuNavigation from './MenuNavigation'

type MobileMenuProps = RouteComponentProps<{}>

type State = {
  open: boolean
}

const Wrap = styled.div`
  padding-left: 40px;
  padding-right: 40px;

  @media (min-width: 1025px) {
    display: none;
  }
`

export class MobileMenu extends React.Component<MobileMenuProps, State> {
  state = {
    open: false,
  }

  componentDidUpdate(prevProps: MobileMenuProps) {
    if (prevProps.location.pathname !== this.props.location.pathname) {
      this.setState({ open: false })
    }
  }

  toggleMenu = () => {
    this.setState(state => ({
      open: !state.open,
    }))
  }

  render() {
    const { open } = this.state

    return (
      <Wrap>
        <MenuNavigation open={open} />
        <MenuButton open={open} toggle={this.toggleMenu} />
      </Wrap>
    )
  }
}

export default withRouter(MobileMenu)
