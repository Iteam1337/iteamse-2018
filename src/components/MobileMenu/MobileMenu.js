// @flow

import * as React from 'react'
import { withRouter, type ContextRouter } from 'react-router-dom'
import styled from 'styled-components'
import MenuButton from './MenuButton'
import MenuNavigation from './MenuNavigation'

type Props = {
  ...ContextRouter,
}

type State = {
  open: boolean,
}

const Wrap = styled.div`
  @media (min-width: 1024px) {
    display: none;
  }
`

class MobileMenu extends React.Component<Props, State> {
  props: Props

  state = {
    open: false,
  }

  componentDidUpdate (prevProps) {
    if (prevProps.location.pathname !== this.props.location.pathname) {
      this.setState({ open: false })
    }
  }

  toggleMenu = () => {
    this.setState(state => ({
      open: !state.open,
    }))
  }

  render () {
    const { open } = this.state

    return (
      <Wrap>
        <MenuNavigation open={open} />
        <MenuButton open={open} toggle={this.toggleMenu} />
      </Wrap>
    )
  }
}

export default withRouter(MobileMenu)
