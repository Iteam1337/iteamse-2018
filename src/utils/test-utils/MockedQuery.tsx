import { InMemoryCache } from 'apollo-cache-inmemory'
import ApolloClient from 'apollo-client'
import * as React from 'react'
import { ApolloProvider } from 'react-apollo'
import { mockSingleLink } from 'react-apollo/test-utils'
import MockRouter from 'react-mock-router'
import { ThemeProvider } from 'styled-components'
import { theme } from '../../theme'

interface MockedQueryProps {
  location?: object
  mocks: any[]
  params?: object
}

const MockedQuery: React.SFC<MockedQueryProps> = ({
  children,
  location,
  mocks,
  params,
}) => {
  const link = mockSingleLink.apply(null, mocks)

  const client = new ApolloClient({
    cache: new InMemoryCache(),
    link,
  })

  return (
    <ApolloProvider client={client}>
      <ThemeProvider theme={theme}>
        <MockRouter location={location} params={params}>
          {children}
        </MockRouter>
      </ThemeProvider>
    </ApolloProvider>
  )
}

export default MockedQuery
