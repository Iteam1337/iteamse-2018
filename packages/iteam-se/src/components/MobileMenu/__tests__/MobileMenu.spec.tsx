import { queries } from 'dom-testing-library'
import 'dom-testing-library/extend-expect'
import React from 'react'
import { cleanup, fireEvent, render, wait } from 'react-testing-library'
import MockedQuery from '../../../utils/test-utils/MockedQuery'
import MobileMenu from '../MobileMenu'

afterEach(cleanup)

describe('components/MobileMenu', () => {
  const mocks = []

  it('renders MobileMenu and can open it', async () => {
    window.resizeTo(411, 500)

    const { getByText, getByTestId, container } = render(
      <MockedQuery location={{ pathname: '/' }} mocks={mocks}>
        <MobileMenu />
      </MockedQuery>
    )

    // Mobile menu should not be open
    expect(queries.queryByText(container, 'Om')).not.toBeInTheDocument()

    // Assert that we have a mobile button and click it
    await wait(() => getByTestId('btn-mobile-menu'))
    fireEvent.click(getByTestId('btn-mobile-menu'))

    // Mobile menu should be open
    expect(getByText('Om')).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })

  it('should close the menu when navigating', async () => {
    const { getByText, getByTestId, container } = render(
      <MockedQuery location={{ pathname: '/' }} mocks={mocks}>
        <MobileMenu />
      </MockedQuery>
    )

    // Assert that we have a mobile button and click it
    await wait(() => getByTestId('btn-mobile-menu'))
    fireEvent.click(getByTestId('btn-mobile-menu'))

    // Mobile menu should be open
    expect(getByText('Om')).toBeInTheDocument()

    // Mock path transition since we don't have React router
    render(
      <MockedQuery location={{ pathname: '/om-oss' }} mocks={mocks}>
        <MobileMenu />
      </MockedQuery>,
      { container }
    )

    // Mobile menu should not be open
    expect(queries.queryByText(container, 'Om')).toMatchSnapshot()
  })
})
