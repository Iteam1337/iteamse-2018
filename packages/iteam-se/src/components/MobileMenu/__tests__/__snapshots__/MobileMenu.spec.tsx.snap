// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MobileMenu renders MobileMenu and can open it 1`] = `
<div>
  <div
    class="MobileMenu__Wrap-sc-1k7u490-0 cYrXHt"
  >
    <div
      data-test="mobile-menu"
      class="MenuNavigation__Menu-unfaa8-0 DOVqg"
    >
      <div
        class="MenuNavigation__MenuInner-unfaa8-1 gOpHSC"
      >
        <a
          class="sc-bdVaJa knFYzh"
          aria-current="false"
        >
          Erbjudanden
        </a>
        <a
          class="sc-bdVaJa knFYzh"
          aria-current="false"
        >
          Våra case
        </a>
        <a
          class="sc-bdVaJa knFYzh"
          aria-current="false"
        >
          Metod
        </a>
        <a
          class="sc-bdVaJa knFYzh"
          aria-current="false"
        >
          Medarbetare
        </a>
        <a
          class="sc-bdVaJa knFYzh"
          aria-current="false"
        >
          Karriär
        </a>
        <a
          class="sc-bdVaJa knFYzh"
          aria-current="false"
        >
          Om
        </a>
        <a
          target="_blank"
          rel="noopener noreferrer"
          href="https://blog.iteam.life/"
          class="NativeLink-fvundc-0 sc-bwzfXH eYkjRL"
        >
          Labs
        </a>
      </div>
    </div>
    <button
      aria-label="Menu button"
      data-testid="btn-mobile-menu"
      role="button"
      class="MenuButton__Button-ulut0i-0 hPPslc"
    >
      <div
        class="MenuButton__Hamburger-ulut0i-2 cykepV"
        open=""
      >
        <div
          class="MenuButton__Bar-ulut0i-1 Bqhsd"
        />
        <div
          class="MenuButton__Bar-ulut0i-1 Bqhsd"
        />
        <div
          class="MenuButton__Bar-ulut0i-1 Bqhsd"
        />
      </div>
    </button>
  </div>
</div>
`;

exports[`components/MobileMenu should close the menu when navigating 1`] = `null`;
